src/frontend/AppContainerWrapper.tsx

```diff
@@ -1,189 +1,105 @@
-}
+  React.useEffect(() => {
+    if (featureFlagOn) {
+      setInitialNavState(undefined);
+      return;
+    }
 
-function hidePracticeModeTemporarily(route: string | null) {
-  return route !== null && TEMP_HIDE_PRACTICE_MODE_UI.includes(route);
-}
+    restoreNavState();
 
-const AppContainerWrapper = () => {
-  const navRef = React.useRef<NavigationContainerComponent>();
-  const [inviteModalEnabled, setInviteModalEnabled] = React.useState(true);
-  const [queuedInvite, setQueuedInvite] = React.useState<string | null>(null);
-  const [hidePracticeBar, setHidePracticeBar] = React.useState(true);
-  const [hidePracticeMode, setHidePracticeMode] = React.useState(false);
-
-  const updateRouteBasedAppState = React.useCallback(
-    (routeName: string | null) => {
-      setInviteModalEnabled(!inviteModalDisabledOnRoute(routeName));
-      setHidePracticeBar(hidePracticeBarForRoute(routeName));
-      setHidePracticeMode(hidePracticeModeTemporarily(routeName));
-    },
-    []
-  );
+    async function restoreNavState() {
+      try {
+        // When we support deeplinking we need to fix this. Currently it is never resolving
+        // You can find issue here: https://github.com/facebook/react-native/issues/25675
+        const initialUrl = undefined; //await Linking.getInitialURL();
 
-  const onNavStateChange = React.useCallback(
-    (previousState: NavigationState, currentState: NavigationState) => {
-      const previousRouteName = getRouteName(previousState);
-      const currentRouteName = getRouteName(currentState);
+        // if deeplinking, dont set initial state
+        if (initialUrl) {
+          setInitialNavState(undefined);
+          return;
+        }
 
-      if (previousRouteName !== currentRouteName) {
-        updateRouteBasedAppState(currentRouteName);
+        setInitialNavState(await loadSavedNavState(log));
+      } catch {
+        // handle error here
       }
-    },
-    [updateRouteBasedAppState]
-  );
-
-  const openInviteModal = React.useCallback((key: string) => {
-    if (navRef.current) {
-      navRef.current.dispatch(
-        NavigationActions.navigate({
-          routeName: "ProjectInviteModal",
-          params: { invite: key },
-        })
-      );
     }
   }, []);
 
-  const { loadNavigationState, persistNavigationState } = React.useMemo(
-    () =>
-      IS_E2E
-        ? {}
-        : {
-            loadNavigationState: async () => {
-              const loadedNavState = await loadSavedNavState();
-
-              const loadedRouteName = getRouteName(loadedNavState);
-
-              updateRouteBasedAppState(loadedRouteName);
-
-              return loadedNavState;
-            },
-            persistNavigationState: createNavigationStatePersister(),
-          },
-    [updateRouteBasedAppState]
+  const handleNavStateChange = React.useCallback(
+    (navState?: NavigationState) => {
+      if (!navState || IS_E2E) return;
+      setCurrentRoute(getRouteName(navState));
```

Comment:
Rather than doing this in a callback, can we not do, in the main body of the function:

```js
const currentRoute = getRouteName(navRef?.getRootState())
```

this way we avoid the extra render (caused by the set state) for every navigation change.  Could also use the `navRef.getCurrentRoute()` method, might be simpler.